core/cautils/fileutils.go

```diff
@@ -190,8 +191,8 @@ func readYamlFile(yamlFile []byte) ([]workloadinterface.IMetadata, error) {
 		}
 		if obj, ok := j.(map[string]interface{}); ok {
 			if o := objectsenvelopes.NewObject(obj); o != nil {
-				if o.GetKind() == "List" {
-					yamlObjs = append(yamlObjs, handleListObject(o)...)
+				if list, ok := o.(*listworkloads.ListWorkloads); ok && list != nil {
```

Comment:
You can use the object type function

Reply:
The NewObject function returns Imetadata object.
The handleListObject function received listWorkloads object.
So in any case it needs to be converted to listWorkloads. And the conversion itself makes the check that it is a list object